plugin.video.sarpur/sarpur/actions.py

```diff
@@ -18,94 +21,60 @@ def index():
+    INTERFACE.add_dir(u'RÚV 2', 'view_category', 'ruv2')
+    INTERFACE.add_dir(u'RÁS 1', 'view_category', 'ras1')
+    INTERFACE.add_dir(u'RÁS 2', 'view_category', 'ras2')
+    INTERFACE.add_dir(u'RÁS 3', 'view_category', 'ras3')
     INTERFACE.add_dir(u'Rondó', 'view_category', 'rondo')
-    INTERFACE.add_dir(u'Krakkasarpurinn', 'view_category', 'born')
     INTERFACE.add_dir(u'Hlaðvarp', 'view_podcast_index', '')
     INTERFACE.add_dir(u'Leita', 'search', '')
 
-def live_index():
-    """
-    List of available live streams
-    """
-    INTERFACE.add_item(u'RÚV', 'play_live', '1')
-    INTERFACE.add_item(u'RÚV Íþróttir', 'play_live', '10')
-    INTERFACE.add_item(u'RÁS 1', 'play_live', '2')
-    INTERFACE.add_item(u'RÁS 2', 'play_live', '3')
-    INTERFACE.add_item(u'Rondó', 'play_live', 'rondo')   
 
-def play_url(url, name):
+def view_live_index(action_value, name):
     """
-    Play media on page (scrapes it to find it)
-
-    :param url: The page url
-    :param name: Text to display in media player
+    List of available live streams
     """
-    video_url = scraper.get_media_url(url)
-    if video_url == -1:
-        GUI.info_box(u"Vesen", u"Fann ekki upptöku")
-    else:
-        player.play(video_url, name)
+    INTERFACE.add_item(u'RÚV', 'play_live_stream', 'ruv')
+    INTERFACE.add_item(u'RÚV 2', 'play_live_stream', 'ruv2')
+    INTERFACE.add_item(u'RÁS 1', 'play_live_stream', 'ras1')
+    INTERFACE.add_item(u'RÁS 2', 'play_live_stream', 'ras2')
+    INTERFACE.add_item(u'Rondó', 'play_live_stream', 'rondo')
 
 
 def play_video(file, name):
     """
-    Play video give only the filename
+    Play video given only the filename
 
-    :param file: Media filename
+    :param file: playable file
     :param name: Text to display in media player
     :return:
     """
-    url = u"http://smooth.ruv.cache.is/opid/{0}".format(file)
-    r = requests.head(url)
-
-    if r.status_code != 200:
-        url = u"http://smooth.ruv.cache.is/lokad/{0}".format(file)
-
-    player.play(url, name)
+    player.play(file, name)
 
 
-def play_podcast(url, name):
-    """
-    Plays podcast (or any media really
-
-    :param url: Direct url to the media
-    :param name: Text to display in media player
-    """
-
-    player.play(url, name)
-
-
-def play_live_stream(category_id, name):
+def play_live_stream(channel, name):
     """
     Play one of the live streams.
 
-    :param category_id: The channel/radio station id (see index())
+    :param channel: The channel/radio station
     :param name: Text to display in media player
     """
-    channel_slugs = {
-        "1": "ruv",
-        "10": "ruv2",
-        "2": "ras1",
-        "3": "ras2",
-        "rondo": "ras3"
-    }
-    url = scraper.get_live_url(channel_slugs[category_id])
+    url = scraper.get_live_url(channel)
     if url == -1:
-        GUI.info_box(u"Vesen", u"Fann ekki straum. Reyndu aftur síðar.")
+        GUI.info_box(u"Vesen", u"Fann ekki straum")
```

Comment:
it would be nice if you could move those strings to localized strings (translations) at some point